tests/rptest/tests/e2e_topic_recovery_test.py

```diff
@@ -249,76 +245,104 @@ def test_restore_with_aborted_tx(self, recovery_overrides):
+                                       transaction_abort_rate=0.5,
+                                       msgs_per_transaction=per_transaction,
+                                       debug_logs=True)
+        producer.start()
+        producer.wait()
+        pstatus = producer.produce_status
+        self.logger.info(f"Producer status: {pstatus}")
+        committed_messages = pstatus.acked - pstatus.aborted_transaction_messages
+        assert pstatus.acked == msg_count
+        # At least some were aborted, at least some were committed
+        assert 0 < committed_messages < msg_count
+
+        # Re-use the same node as the producer for consumers
+        traffic_node = producer.nodes[0]
 
-        wait_until(done,
-                   timeout_sec=120,
-                   backoff_sec=1,
-                   err_msg="producing failed")
+        rpk = RpkTool(self.redpanda)
+        hwm = next(rpk.describe_topic(self.topic)).high_watermark
+        self.logger.info(f"Produced to HWM {hwm}")
 
-        assert producer.num_aborted > 0
+        # Each transaction is amplified with one tx start and one tx end marker
+        assert hwm == msg_count + (msg_count / per_transaction) * 2
 
-        time.sleep(10)
-        wait_until(lambda: self._s3_has_all_data(producer.cur_offset),
-                   timeout_sec=300,
+        wait_until(lambda: self._s3_has_all_data(hwm),
+                   timeout_sec=600,
                    backoff_sec=5,
                    err_msg=f"Not all data is uploaded to S3 bucket")
 
-        # Whipe out the state on the nodes
+        # Keep services alive, so that log-capturing gets their logs at the end
+        consumers = []
+
+        def validate():
+            consumer = KgoVerifierSeqConsumer(self.test_context,
+                                              self.redpanda,
+                                              self.topic,
+                                              msg_size,
+                                              loop=False,
+                                              nodes=[traffic_node],
+                                              debug_logs=True,
+                                              trace_logs=True)
+            consumer.start(clean=False)
+            consumers.append(consumer)
+            consumer.wait()
+            status = consumer.consumer_status
+            self.logger.info(f"Consumer status: {status}")
+            try:
+                assert status.errors == 0
+                assert status.validator.valid_reads == committed_messages
+
+                # No aborted messages should be visible
+                assert status.validator.invalid_reads == 0
+
+                # No junk on the topic outside of the ranges our producer produced
+                assert status.validator.out_of_scope_invalid_reads == 0
+            except Exception:
+                self.logger.exception("I need an adult")
+                time.sleep(3600)
```

Comment:
Why the long pause? Does the process actually continue if we log an exception? Maybe add a comment since it's tempting to remove (or remove it if this was unintentional?)

Reply:
Oops, good catch, this is my debugging hack for running locally.  Will remove.